src/Ookii.Dialogs.Wpf/ProgressDialog.cs

```diff
@@ -680,19 +693,28 @@ public void ReportProgress(int percentProgress, string text, string description,
         protected virtual void OnDoWork(DoWorkEventArgs e)
         {
             DoWorkEventHandler handler = DoWork;
-            if (!(handler is null))
+            if (handler is null)
             {
-                var eventArgs = new ProgressDialogDoWorkEventArgs(e.Argument, CancellationToken.None)
+                return;
+            }
+
+            using (var cts = System.Threading.CancellationTokenSource.CreateLinkedTokenSource(_cancellationToken))
+            {
+                var eventArgs = new ProgressDialogDoWorkEventArgs(e.Argument, cts.Token)
                 {
                     Cancel = e.Cancel,
                     Result = e.Result,
                 };
 
+                _cancellationSource = cts;
+
                 handler(this, eventArgs);
 
+                _cancellationSource = null;
```

Comment:
Same as above. This should go. We dispose the `CancellationTokenSource` after the Background worker finished (successfully or not)